index.js

```diff
@@ -1,88 +1,130 @@
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
+
+
+const sameSameButDifferent = (num, maybeNum) => {
+  if (num === maybeNum) {return "same same"}
+  else if (num == maybeNum) {return "same same (but different)"}
+  else {return "different"}
+}
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
+// * The solution below more literally fits the directions  *
+// const sameSameButDifferent = (num, maybeNum) => {
+//   if ((typeof(num) && typeof(maybeNum)) == "number" && num === maybeNum){
+//     return "same same"
+//   }
+//   else if ((typeof(num) != typeof(maybeNum)) && parseFloat(num) === parseFloat(maybeNum)) {
+//     return "same same (but different)"
+//   }
+//   else{
+//     return "different"
+//   }
+// }
+
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
-// // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+console.log("%cQuestion 5", "color: red")
+const student1 = { name: "Duane", grade: 88 }
+
+const updateGrade = (student, grade) => {
+  student.grade = grade
+  return student
+}
+
+updateGrade(student1, 92)
+console.log(student1)
+// => { name: "Duane", grade: 92 }
+console.log("%c----------", "color: red")
 
 
 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
-// // => "Duane"
+console.log("%cQuestion 6", "color: red")
+
+const users = [ 
+  { 
+    name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
+  },
+  { 
+    name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
+  }
+]
+
+const printNameAndPhones = peeps => {
+  for (const user of peeps){
+    console.log(user.name)
+    console.log(`Cell: ${user.phones.cell}`)
+    console.log(`Office: ${user.phones.office}`)
+  }
+}
+printNameAndPhones(users)
```

Comment:
Great! - Here's how it would look with forEach:
```javascript
 users.forEach(function(user) {
  console.log(user.name);
  console.log(`Cell: ${user.phones.cell}`);
  console.log(`Office: ${user.phones.office}`);
 })
```